pulpcore/pulpcore/app/views/content.py

```diff
@@ -104,87 +148,127 @@ def _match(self, path):
         """
-        pass
-        # :TODO:
+        response = HttpResponse()
+        response['X-SENDFILE'] = path
+        return response
 
-    def _apache(self, storage_path):
+    def _nginx(self, path):
         """
-        The content web server is Apache.
+        The content web server is NGINX.
 
         Args:
-            storage_path (str): The storage path of the requested object.
+            path (str): The fully qualified path to the file to be served.
 
         Returns:
-            HttpResponse: A response with X-SENDFILE header.
+            HttpResponse: A response with X-Accel-Redirect header.
         """
         response = HttpResponse()
-        response['X-SENDFILE'] = storage_path
+        response['X-Accel-Redirect'] = path
         return response
 
-    def _nginx(self, storage_path):
+    def _redirect(self, request):
         """
-        The content web server is NGINX.
+        Get redirect-to-streamer response.
 
         Args:
-            storage_path (str): The storage path of the requested object.
+            request (django.http.HttpRequest): A request for a content artifact.
 
         Returns:
-            HttpResponse: A response with X-Accel-Redirect header.
+            HttpResponseRedirect: Redirect to streamer.
         """
-        response = HttpResponse()
-        response['X-Accel-Redirect'] = storage_path
+        redirect = settings.CONTENT['REDIRECT']
+
+        enabled = redirect['ENABLED']
+        host = redirect['HOST'] or request.get_host().split(':')[0]
+        port = redirect['PORT'] or request.get_port()
+        prefix = redirect['PATH_PREFIX']
+
+        if not enabled:
+            return HttpResponseNotFound()
+
+        path = os.path.join(
+            prefix.strip('/'),
+            self._published_path(request)
+        )
+
+        url = '{scheme}://{host}:{port}/{path}'.format(
+            scheme=request.scheme,
+            host=host,
+            path=path,
+            port=port)
+
+        log.debug(_('Redirected: %(u)s'), {'u': url})
+        response = HttpResponseRedirect(url)
         return response
 
-    # Mapping of responder method by web server.
-    RESPONDER = {
-        'django': _stream,
-        'apache': _apache,
-        'nginx': _nginx,
-    }
+    def _dispatch(self, path):
+        """
+        Dispatch to the appropriate responder (method).
+
+        Args:
+            path (str): The fully qualified path to the file to be served.
+
+        Returns:
+            django.http.StreamingHttpResponse: on found.
+            django.http.HttpResponseNotFound: on not-found.
+            django.http.HttpResponseForbidden: on forbidden.
+            django.http.HttpResponseRedirect: on redirect to the streamer.
+        """
+        server = settings.CONTENT['WEB_SERVER']
+
+        try:
+            responder = self.RESPONDER[server]
+        except KeyError:
+            raise ValueError(_('Web server "{t}" not supported.'.format(t=server)))
```

Comment:
I'm not able to do a full review right now, but I noticed this line. The .format() needs to run on str returned from _() not the string passed into _(). This won't i8ln as is.

Reply:
Fixed.